app/controllers/articles_controller.rb

```diff
@@ -0,0 +1,53 @@
+class ArticlesController < ApplicationController
+  
+  http_basic_authenticate_with name: "user", password: "secret", except: [:index, :show]
+
+  def index
+    @articles = Article.all
+  end
+
+  def show
+    @article = Article.find(params[:id])
```

Comment:
What happens if there is no `Article` with the specified `id`? What should you do if that's the case?

Reply:
I reviewed this situation and throws an error when id is not found. In this case the solution is to render the 404-page error.

Reply:
```
class ApplicationController < ActionController::Base
  def not_found
    raise ActionController::RoutingError.new('Not Found')
  end
end
```